### PR TITLE
wait_for_finish method, which waits for pipeline run to finish, after we

### PR DESCRIPTION
**Pull request by rcerven**
finished reading logs, there is race between all pods finished and
pipeline run changing status

we will wait just for certain amount of time, if pipeline run still
didn't finish by then, there is something really wrong and that will be
already handled

* CLOUDBLD-8359

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a] JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

**Review by hjmodi — APPROVED** [review]

**Review by lkolacek — APPROVED** [review]
Code change looks good to me. Have you captured this race condition issue? Can you post more info to CLOUDBLD-8359?

File: osbs/tekton.py [hunk 1]
```diff
@@ -556,6 +559,16 @@ def has_not_finished(self):
...
 
+    def wait_for_finish(self):
+        for _ in range(WAIT_RETRY):
+            if self.has_not_finished():
+                logger.info("Waiting for pipeline run '%s' to finish, sleep for %ss",
+                            self.pipeline_run_name, WAIT_RETRY_SECS)
+                time.sleep(WAIT_RETRY_SECS)
+            else:
+                logger.info("Pipeline run '%s' finished", self.pipeline_run_name)
+                break
```
Comment (chmeliik) [inline osbs/tekton.py:570]: Should we raise an exception if the loop exits normally (i.e. the pipeline does not finish within the timeout)?

File: osbs/tekton.py [hunk 2]
```diff
@@ -556,6 +559,16 @@ def has_not_finished(self):
     def was_cancelled(self):
         return self.status_reason == 'PipelineRunCancelled'
 
+    def wait_for_finish(self):
```
Comment (chmeliik) [inline osbs/tekton.py:562]: I would appreciate a docstring that explains how this method should be used - only for "almost finished" pipeline runs, not newly started ones.

File: osbs/tekton.py (hunk 1 above)
Comment (rcerven) [reply osbs/tekton.py:570]: there isn't any benefit in that, we will then later anyway call build_has_succeeded and build_not_finished (and cancel)

this is really just for getting some time for pipeline run to finish

File: osbs/tekton.py (hunk 2 above)
Comment (rcerven) [reply osbs/tekton.py:562]: added

File: osbs/tekton.py (hunk 1 above)
Comment (chmeliik) [reply osbs/tekton.py:574]: Ok, as long as the purpose of the function is documented this seems fine

**Review by chmeliik — APPROVED** [review]

**Comment by rcerven** [comment]
no rn update, fix for unreleased changes

**Pull request merged**